src/org/intellij/erlang/psi/impl/ErlangPsiImplUtil.java

```diff
@@ -836,6 +836,42 @@ public static ErlangFile resolveToFile(@Nullable ErlangModuleRef ref) {
     return ObjectUtils.tryCast(resolved != null ? resolved.getContainingFile() : null, ErlangFile.class);
   }
 
+  @Nullable
+  public static ErlangFunction resolveToFunction(@Nullable ErlangFunctionCallExpression call) {
+    PsiReference reference = call != null ? call.getReference() : null;
+    PsiElement resolvedFunction = reference != null ? reference.resolve() : null;
+    return ObjectUtils.tryCast(resolvedFunction, ErlangFunction.class);
+  }
+
+  @NotNull
+  public static List<ErlangFunction> resolveToFunctions(@NotNull ErlangFunctionCallExpression call) {
+    PsiReference reference = call.getReference();
+    ResolveResult[] results = reference instanceof PsiPolyVariantReference ?
+                              ((PsiPolyVariantReference) reference).multiResolve(true) : ResolveResult.EMPTY_ARRAY;
+    return ContainerUtil.mapNotNull(results, new Function<ResolveResult, ErlangFunction>() {
+      @Override
+      @Nullable
+      public ErlangFunction fun(ResolveResult resolveResult) {
+        return ObjectUtils.tryCast(resolveResult.getElement(), ErlangFunction.class);
+      }
+    });
+  }
+
+  @Nullable
+  public static ErlangExpression getFunctionCallArgument(@NotNull ErlangFunctionCallExpression call,
+                                                         @Nullable PsiElement child) {
+    ErlangExpression argument = PsiTreeUtil.getParentOfType(child, ErlangExpression.class, false);
+    while (argument != null && !(argument.getParent() instanceof ErlangArgumentList)) {
+      argument = ObjectUtils.tryCast(argument.getParent(), ErlangExpression.class);
+    }
+    return argument != null && call.getArgumentList() != argument.getParent()
+           ? getFunctionCallArgument(call, argument.getParent()) : argument;
+  }
```

Comment:
Does this function have to be that complex? You given a function call expression, and a psi element that is supposed to be a child of an argument of this call - why all these casts, instanceof checks, recursion?